source/shared/cpp/ObjectModel/Enums.h

```diff
@@ -371,7 +371,8 @@ namespace AdaptiveSharedNamespace
         RequiredPropertyMissing,
         InvalidPropertyValue,
         UnsupportedParserOverride,
-        IdCollision
+        IdCollision,
+        CustomError,
```

Comment:
>CustomError [](start = 8, length = 11)

Nit: CustomError sounds odd ? Would Default be better? (Is this similar to an E_FAIL?

Reply:
This basically means that the error is not one of the known errors returned by the parser/renderer, and is a "custom" error provided by, for example, a custom parser. I chose "Custom" thinking of consistency with the Custom enum values in CardElementType and ActionType.

As implemented, it's not really "Default" in any meaningful way (the error object doesn't default to that value, for example). Default error or warning code doesn't necessarily seem less odd to me (what does a "default" warning mean to someone consuming it?), but perhaps "Other" or "Unknown"? I think i still like "Custom" best, as it seems to me to convey the information that this isn't a known type but has been explicitly customized (i.e., there's likely something interesting in the message). What do you think?

Thinking about defaults though, i wonder if the constructor should require an error/warning status code to be passed in to ensure a meaningful object here...

---
In reply to: [316942726](https://github.com/microsoft/AdaptiveCards/pull/3375#discussion_r316942726) [](ancestors = 316942726)

Reply:
FWIW, I'm in agreement on `CustomX` here for consistency reasons. Implementing a "custom" renderer and a "custom" parser feels like it might require "custom" errors and warnings.

---
In reply to: [317235188](https://github.com/microsoft/AdaptiveCards/pull/3375#discussion_r317235188) [](ancestors = 317235188,316942726)

Reply:
I did update the constructor to require status code and string. Shared model already does this and as we don't actually have a default value, requiring one to be passed in makes sense.

I'm going to stick with CustomX for now. @shalinijoshi19 feel free to follow up when you're back if you have further concerns.

---
In reply to: [317236272](https://github.com/microsoft/AdaptiveCards/pull/3375#discussion_r317236272) [](ancestors = 317236272,317235188,316942726)